pdc_client/plugins/product.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+
+from __future__ import print_function
+
+import json
+try:
+    from collections import OrderedDict
+except ImportError:
+    # Python 2.6 needs this back-port
+    from ordereddict import OrderedDict
+
+from pdc_client.plugin_helpers import (PDCClientPlugin,
+                                       extract_arguments,
+                                       add_create_update_args)
+
+
+class ProductPlugin(PDCClientPlugin):
+    def register(self):
+        self.set_command('product')
+
+        # CRUD: CREATE
+        create_parser = self.add_action('create', help='create new product')
+        self.add_product_arguments(create_parser, required=True)
+        create_parser.set_defaults(func=self.product_create)
+
+        # CRUD: READ one (info)
+        info_parser = self.add_action('info', help='display details of a product')
+        info_parser.add_argument('short', metavar="SHORT")
+        info_parser.set_defaults(func=self.product_info)
+
+        # CRUD: READ many (list)
+        list_parser = self.add_action('list', help='list all products')
```

Comment:
According to the 'product' API, the parameters of 'name' and 'short'  will be added to filter the results.
Reference: https://pdc.engineering.redhat.com/rest_api/v1/products/#tab_list .